spec/controllers/users_controller_spec.rb

```diff
@@ -1,88 +1,84 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe Clearance::UsersController do
   it { should be_a Clearance::BaseController }
 
-  describe 'when signed out' do
-    before { sign_out }
+  describe "on GET to #new" do
+    context "when signed out" do
+      context "with email" do
+        it "should set assigned user email" do
+          sign_out
 
-    describe 'on GET to #new' do
-      before { get :new }
+          get :new, user: { email: "a@example.com" }
 
-      it { is_expected.to respond_with(:success) }
-      it { is_expected.to render_template(:new) }
-      it { is_expected.not_to set_the_flash }
-    end
-
-    describe 'on GET to #new with email' do
-      before do
-        @email = 'a@example.com'
-        get :new, user: { email: @email }
+          expect(assigns(:user).email).to eq "a@example.com"
+        end
       end
 
-      it 'should set assigned user email' do
-        expect(assigns(:user).email).to eq @email
+      context "without email" do
+        it "renders the :new template without setting the flash" do
+          sign_out
+
+          get :new
+
+          expect(response).to be_success
+          expect(response).to render_template(:new)
+          expect(flash).to be_blank
+        end
       end
     end
 
-    describe 'on POST to #create with valid attributes' do
-      before do
-        user_attributes = FactoryGirl.attributes_for(:user)
-        @old_user_count = User.count
-        post :create, user: user_attributes
-      end
+    context "when signed in" do
+      it "redirects to the redirect_url" do
+        sign_in
 
-      it 'assigns a user' do
-        expect(assigns(:user)).to be_present
-      end
+        get :new
 
-      it 'should create a new user' do
-        expect(User.count).to eq @old_user_count + 1
+        expect(response).to redirect_to(Clearance.configuration.redirect_url)
       end
-
-      it { should redirect_to_url_after_create }
     end
+  end
 
-    describe 'on POST to #create with valid attributes and a session return url' do
-      before do
-        user_attributes = FactoryGirl.attributes_for(:user)
-        @old_user_count = User.count
-        @return_url = '/url_in_the_session'
-        @request.session[:return_to] = @return_url
-        post :create, user: user_attributes
-      end
+  describe "on POST to #create" do
+    context "when signed out" do
+      context "with valid attributes" do
+        it "assigns and creates a user then redirects to the redirect_url" do
+          sign_out
```

Comment:
no they are not... they were brought over from the previous `before` blocks... I will remove them